.github/workflows/main.yml

```diff
@@ -140,33 +140,172 @@ jobs:
+        # comma-separated list of test targets for the matrix.
+        run: |
+          TEST_TARGET_LIST=$(bash ./scripts/compute_affected_tests.sh $HOME/oppia-bazel/bazel | sed 's/^\|$/"/g' | paste -sd, -)
+          echo "Affected tests (note that this might be all tests if on the develop branch): $TEST_TARGET_LIST"
+          echo "::set-output name=matrix::{\"test-target\":[$TEST_TARGET_LIST]}"
+      - name: Compute test matrix based on all tests
+        id: compute-test-matrix-from-all
+        if: "contains(github.event.pull_request.title, '[RunAllTests]')"
+        run: |
+          TEST_TARGET_LIST=$($HOME/oppia-bazel/bazel query "kind(test, //...)" | sed 's/^\|$/"/g' | paste -sd, -)
+          echo "Affected tests (note that this might be all tests if on the develop branch): $TEST_TARGET_LIST"
+          echo "::set-output name=matrix::{\"test-target\":[$TEST_TARGET_LIST]}"
+
+  bazel_run_test:
+    name: Run Bazel Test
+    needs: bazel_compute_affected_targets
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix: ${{fromJson(needs.bazel_compute_affected_targets.outputs.matrix)}}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Clone Oppia Bazel
+        run: git clone https://github.com/oppia/bazel.git $HOME/oppia-bazel
+      - name: Set up JDK 9
+        uses: actions/setup-java@v1
+        with:
+          java-version: 9
+      - name: Extract Android tools
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/tmp/android_tools
+          cd $HOME/oppia-bazel
+          unzip bazel-tools.zip
+          tar -xf $HOME/oppia-bazel/android_tools.tar.gz -C $GITHUB_WORKSPACE/tmp/android_tools
+      # See explanation in bazel_build_app for how this is installed.
+      - name: Install git-secret (non-fork only)
+        if: github.repository == 'oppia/oppia-android'
+        shell: bash
+        run: |
+          cd $HOME
+          mkdir -p $HOME/gitsecret
+          git clone https://github.com/sobolevn/git-secret.git git-secret
+          cd git-secret && make build
+          PREFIX="$HOME/gitsecret" make install
+          echo "$HOME/gitsecret" >> $GITHUB_PATH
+          echo "$HOME/gitsecret/bin" >> $GITHUB_PATH
+      - name: Decrypt secrets (non-fork only)
+        if: github.repository == 'oppia/oppia-android'
+        env:
+          GIT_SECRET_GPG_PRIVATE_KEY: ${{ secrets.GIT_SECRET_GPG_PRIVATE_KEY }}
+        run: |
+          cd $HOME
+          # NOTE TO DEVELOPERS: Make sure to never print this key directly to stdout!
+          echo $GIT_SECRET_GPG_PRIVATE_KEY | base64 --decode > ./git_secret_private_key.gpg
+          gpg --import ./git_secret_private_key.gpg
+          cd $GITHUB_WORKSPACE
+          git secret reveal
+      - name: Unzip Bazel binary
+        run: |
+          cd $HOME/oppia-bazel
+          unzip bazel-build.zip
+          cd $GITHUB_WORKSPACE
+          chmod a+x $HOME/oppia-bazel/bazel
```

Comment:
Do we have a modular way, as the cloning oppia/bazel and setting the secret key is something we are performing more than once?

Reply:
Yes, this was added to GitHub recently: https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/. It requires creating custom actions that perform sub-steps in the same repo. I didn't pursue it since it would take more time to figure out how to do this correctly, but I can look into if you feel strongly (I'm also not a fan of reusing large, complicated portions of the actions script, but I also feel that this is something we can simplify in subsequent PRs).

Reply:
Sounds good. This could be a good learning issue around GitHub actions.

Reply:
Good idea--will keep this open as a reminder for me to file an issue for this later.

Reply:
Filed #2692.